tests/persistence/test_persistence_leverage.py

```diff
@@ -0,0 +1,638 @@
+    # Get the open rate price with the standard fee rate
+    assert trade._calc_open_trade_value() == 0.01134051354788177
+    trade.fee_open = 0.003
+    # Get the open rate price with a custom fee rate
+    assert trade._calc_open_trade_value() == 0.011346169664364504
+
+
+@pytest.mark.usefixtures("init_persistence")
+def test_calc_open_close_trade_price_lev(limit_lev_buy_order, limit_lev_sell_order, fee):
+    """
+        5 hour leveraged trade on Binance
+
+        fee: 0.25% base
+        interest_rate: 0.05% per day
+        open_rate: 0.00001099 base
+        close_rate: 0.00001173 base
+        amount: 272.97543219 crypto
+        stake_amount: 0.0009999999999226999 base
+        borrowed: 0.0019999999998453998  base
+        time-periods: 5 hours(rounds up to 5/24 time-period of 1 day)
+        interest: borrowed * interest_rate * time-periods
+                    = 0.0019999999998453998 * 0.0005 * 5/24 = 2.0833333331722917e-07 base
+        open_value: (amount * open_rate) + (amount * open_rate * fee)
+            = (272.97543219 * 0.00001099) + (272.97543219 * 0.00001099 * 0.0025)
+            = 0.0030074999997675204
+        close_value: ((amount_closed * close_rate) - (amount_closed * close_rate * fee)) - interest
+            = (272.97543219 * 0.00001173)
+                - (272.97543219 * 0.00001173 * 0.0025)
+                - 2.0833333331722917e-07
+            = 0.003193788481706411
+        stake_value: (amount/lev * open_rate) + (amount/lev * open_rate * fee)
+            = (272.97543219/3 * 0.00001099) + (272.97543219/3 * 0.00001099 * 0.0025)
+            = 0.0010024999999225066
+        total_profit =  close_value - open_value
+            = 0.003193788481706411 - 0.0030074999997675204
+            = 0.00018628848193889044
+        total_profit_percentage = total_profit / stake_value
+            = 0.00018628848193889054 / 0.0010024999999225066
+            = 0.18582392214792087
+    """
+    trade = Trade(
+        pair='ETH/BTC',
+        stake_amount=0.0009999999999226999,
+        open_rate=0.01,
+        amount=5,
+        open_date=datetime.utcnow() - timedelta(hours=4, minutes=55),
+        fee_open=fee.return_value,
+        fee_close=fee.return_value,
+        exchange='binance',
+        interest_rate=0.0005,
+        interest_mode=InterestMode.HOURSPERDAY
+    )
+    trade.open_order_id = 'something'
+    trade.update(limit_lev_buy_order)
+    assert trade._calc_open_trade_value() == 0.00300749999976752
+    trade.update(limit_lev_sell_order)
+
+    # Is slightly different due to compilation time changes. Interest depends on time
+    assert round(trade.calc_close_trade_value(), 11) == round(0.003193788481706411, 11)
+    # Profit in BTC
+    assert round(trade.calc_profit(), 8) == round(0.00018628848193889054, 8)
+    # Profit in percent
+    assert round(trade.calc_profit_ratio(), 8) == round(0.18582392214792087, 8)
+
+
+@pytest.mark.usefixtures("init_persistence")
+def test_trade_close_lev(fee):
```

Comment:
I think this test would offer itself for a parametrization - adding leverage as parameter (use 0/1, 3, 5? ) - and then evaluating the different results (obviously more leverage = more profit in case of wins)

might even make sense to do the same with losing trades (make sure the calculation is correct there too).

Reply:
Just a heads up, the default will be 1x leverage, 0x leverage will mean 0 * amount = 0